device/nexthop/x86_64-nexthop_4010-r0/pddf/pddf-device.json

```diff
@@ -0,0 +1,13423 @@
+{
+  "PLATFORM": {
+    "num_psus": 2,
+    "num_fantrays": 4,
+    "num_fans_pertray": 2,
+    "num_ports": 66,
+    "num_temps": 6,
+    "num_asic_temps": 56,
+    "num_voltage_sensors" : 24,
+    "num_current_sensors" : 8,
+    "num_nexthop_fpga_asic_temp_sensors": 2,
+    "num_components": 6,
+    "pddf_dev_types": {
+      "description": " - Below is the list of supported PDDF device types (chip names) for various components. If any component uses some other driver, we will create the client using 'echo <dev-address> <dev-type> > <path>/new_device' method",
+      "MULTIFPGAPCIE": [
+        "multifpgapci"
+      ],
+      "PORT_MODULE": [
+        "pddf_xcvr"
+      ],
+      "FAN": [
+        "fan_multifpgapci"
+      ],
+      "PSU": [
+        "psu_pmbus"
+      ]
+    },
+    "std_kos": [
+      "at24",
+      "i2c-dev",
+      "adm1266",
+      "optoe"
+    ],
+    "pddf_kos": [
+      "pddf_client_module",
+      "pddf_multifpgapci_driver",
+      "pddf_multifpgapci_module",
+      "pddf_led_module",
+      "pddf_psu_module",
+      "pddf_fan_driver_module",
+      "pddf_fan_module",
+      "pddf_xcvr_module",
+      "pddf_xcvr_driver_module",
+      "pddf_cpldmux_module",
+      "pddf_cpldmux_driver"
+    ],
+    "custom_kos": [
+      "pddf_custom_fpga_algo",
+      "nh_pmbus_core",
```

Comment:
Is it necessary to use the 'nh_pmbus_core' driver for the PSUs? 
If pddf_psu_driver can support with small modification to set/get or read/write functions then taht can be explored.

Reply:
Yes, you can use most of the pddf common driver with small modification in the custom driver instead of creating a whole new driver.